### PR TITLE
Fixed#10565: Make Prometheus Mappings Configurable

### LINKED ISSUE #10565: Make Prometheus Mappings Configurable
**Is your feature request related to a problem? Please describe.**
Currently, we are monitoring every API endpoint, we need to have a configurable mechanism to allow users to monitor specific endpoints

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by parthp2107**
### Describe your changes :

Fixes #10565 

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@mohityadav766 @harshach

### CODE REVIEW

File: conf/openmetadata.yaml [hunk 1]
```diff
@@ -199,6 +199,7 @@ elasticsearch:
 eventMonitoringConfiguration:
   eventMonitor: ${EVENT_MONITOR:-prometheus}  # Possible values are "prometheus", "cloudwatch"
   batchSize: ${EVENT_MONITOR_BATCH_SIZE:-10}
+  pathPattern: ${EVENT_MONITOR_PATH_PATTERN:-["/*"]}
```
Comment (harshach) [inline conf/openmetadata.yaml:202]: we should also what metrics we want to monitor, like P90 , p95, p99 etc.. cc @ulixius9

Comment (parthp2107) [reply conf/openmetadata.yaml:202]: Okay, will have a look into it.

Comment (harshach) [reply conf/openmetadata.yaml:202]: also lets add few endPoints we want to monitor
/api/v1/tables
/api/v1/healthcheck
cc @mohityadav766

Comment (parthp2107) [reply conf/openmetadata.yaml:202]: Sure, thank you for the feedback.

Comment (ynaguib) [reply conf/openmetadata.yaml:202]: Is it possible to also have an exclusion pattern?
Ex: EVENT_MONITOR_EXCLUDE_PATH_PATTERN

File: conf/openmetadata.yaml [hunk 2]
```diff
@@ -225,6 +225,8 @@ elasticsearch:
 eventMonitoringConfiguration:
   eventMonitor: ${EVENT_MONITOR:-prometheus}  # Possible values are "prometheus", "cloudwatch"
   batchSize: ${EVENT_MONITOR_BATCH_SIZE:-10}
+  pathPattern: ${EVENT_MONITOR_PATH_PATTERN:-["/*", "/api/v1/tables/*", "/api/v1/health-check"]}
```
Comment (mohityadav766) [inline conf/openmetadata.yaml:228]: if we have "/*" then mapping for /api/v1/tables/*", "/api/v1/health-check is by default there , remove /* 
@harshach any other endpoint we want to monitor

File: openmetadata-service/src/main/java/org/openmetadata/service/security/saml/OMMicrometerHttpFilter.java [hunk 3]
```diff
@@ -39,16 +41,23 @@ public void init(FilterConfig filterConfig) throws ServletException {
...
-    }
-    String responseStatus = String.valueOf(((HttpServletResponse) response).getStatus());
-    String requestMethod = ((HttpServletRequest) request).getMethod();
-    MicrometerBundle.httpRequests.labels(requestMethod, responseStatus, requestPath).observe(elapsed);
+    webAnalyticEvents.record(
+        () -> {
+          long startTime = System.nanoTime();
+          try {
+            chain.doFilter(request, response);
+          } catch (IOException | ServletException e) {
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/security/saml/OMMicrometerHttpFilter.java:49]: what are we doing here ?? the method already throws IOException and ServletException , why we need to catch it and throw RuntimeException?

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**